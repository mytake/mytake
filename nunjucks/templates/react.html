{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
