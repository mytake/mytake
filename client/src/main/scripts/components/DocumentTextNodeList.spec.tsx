/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2018 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import DocumentTextNodeList from "./DocumentTextNodeList";
import { documentNodes } from "../utils/testUtils";

const onMouseUp = jest.fn();
const className = "document__text";

jest.mock("./DocumentTextNode", () => ({
  __esModule: true,
  default: "DocumentTextNode",
}));

test("DocumentTextNodeList", () => {
  const tree = renderer
    .create(
      <DocumentTextNodeList
        documentNodes={documentNodes}
        onMouseUp={onMouseUp}
        className={className}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
