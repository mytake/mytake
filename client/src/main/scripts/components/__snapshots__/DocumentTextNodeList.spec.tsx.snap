// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentTextNodeList 1`] = `
<div
  className="document__text"
  onMouseUp={[MockFunction]}
>
  <DocumentTextNode
    documentNode={
      Object {
        "component": "p",
        "innerHTML": Array [
          "Section 1. Neither slavery nor involuntary servitude, except as a punishment for crime whereof the party shall have been duly convicted, shall exist within the United States, or any place subject to their jurisdiction.",
        ],
        "offset": 0,
      }
    }
  />
  <DocumentTextNode
    documentNode={
      Object {
        "component": "p",
        "innerHTML": Array [
          "Section 2. Congress shall have power to enforce this article by appropriate legislation.",
        ],
        "offset": 218,
      }
    }
  />
</div>
`;
